File: scripts/add-app-layout.py
"""
Adds AppLayout wrapper to internal pages missing it.
Uses Python for reliable multi-line import detection.
"""
import re
import os

PAGES_DIR = "client/src/pages"
IMPORT_LINE = 'import AppLayout from "@/components/AppLayout";\n'

PAGES_TO_FIX = [
    "EmailSequences.tsx",
    "CRM.tsx",
    "Invoicing.tsx",
    "SMS.tsx",
    "Social.tsx",
    "Automations.tsx",
    "Team.tsx",
    "Reputation.tsx",
    "Booking.tsx",
    "Contracts.tsx",
    "LinkedInOutreach.tsx",
    "VoiceAgents.tsx",
    "Funnels.tsx",
]

def find_last_import_end(lines):
    """Find the line index AFTER the last import statement (handles multi-line imports)."""
    last_import_end = -1
    i = 0
    while i < len(lines):
        line = lines[i]
        if line.startswith("import "):
            # Check if this is a multi-line import (has { but no })
            if "{" in line and "}" not in line:
                # Find the closing }
                while i < len(lines) and "}" not in lines[i]:
                    i += 1
            last_import_end = i + 1
        i += 1
    return last_import_end

def find_main_return(lines):
    """Find the last top-level 'return (' in the file (2-space indent)."""
    for i in range(len(lines) - 1, -1, -1):
        if re.match(r'^  return \(', lines[i]):
            return i
    return -1

def find_closing_paren(lines, start):
    """Find the closing ); of a return block starting at `start`."""
    # Walk forward looking for a line that is just '  );' at the right indent
    for i in range(start + 1, len(lines)):
        if re.match(r'^  \);', lines[i]):
            return i
    return -1

for filename in PAGES_TO_FIX:
    filepath = os.path.join(PAGES_DIR, filename)
    if not os.path.exists(filepath):
        print(f"SKIP (not found): {filename}")
        continue

    with open(filepath, "r") as f:
        content = f.read()

    if "AppLayout" in content:
        print(f"SKIP (already has AppLayout): {filename}")
        continue

    lines = content.split("\n")

    # 1. Find where to insert the import (after last import block)
    import_insert_idx = find_last_import_end(lines)
    if import_insert_idx == -1:
        print(f"SKIP (no imports): {filename}")
        continue

    # 2. Insert import line
    lines.insert(import_insert_idx, 'import AppLayout from "@/components/AppLayout";')

    # 3. Find main return ( after import insertion
    return_idx = find_main_return(lines)
    if return_idx == -1:
        print(f"SKIP (no return): {filename}")
        continue

    # 4. Find closing );
    close_idx = find_closing_paren(lines, return_idx)
    if close_idx == -1:
        print(f"SKIP (no closing): {filename}")
        continue

    # 5. Insert </AppLayout> before the closing );
    lines.insert(close_idx, "    </AppLayout>")

    # 6. Insert <AppLayout> after the return (
    lines.insert(return_idx + 1, "    <AppLayout>")

    new_content = "\n".join(lines)
    with open(filepath, "w") as f:
        f.write(new_content)
    print(f"FIXED: {filename}")

print("\nDone.")

File: scripts/add-app-layout.mjs
/**
 * Adds AppLayout wrapper to internal pages that are missing it.
 * Each page gets:
 *   - import AppLayout from "@/components/AppLayout"; (if not present)
 *   - The main return wrapped in <AppLayout>...</AppLayout>
 */
import fs from "fs";
import path from "path";

const PAGES_DIR = path.resolve("client/src/pages");

// Pages to fix: internal app pages that need the nav/header
const PAGES_TO_FIX = [
  "EmailSequences.tsx",
  "CRM.tsx",
  "Invoicing.tsx",
  "SMS.tsx",
  "Social.tsx",
  "Automations.tsx",
  "Team.tsx",
  "Reputation.tsx",
  "Booking.tsx",
  "Contracts.tsx",
  "LinkedInOutreach.tsx",
  "VoiceAgents.tsx",
  "Funnels.tsx",
];

for (const filename of PAGES_TO_FIX) {
  const filePath = path.join(PAGES_DIR, filename);
  if (!fs.existsSync(filePath)) {
    console.log(`SKIP (not found): ${filename}`);
    continue;
  }

  let content = fs.readFileSync(filePath, "utf8");

  // Skip if already has AppLayout
  if (content.includes("AppLayout")) {
    console.log(`SKIP (already has AppLayout): ${filename}`);
    continue;
  }

  // 1. Add AppLayout import after the last import line
  const lastImportIdx = content.lastIndexOf("\nimport ");
  const endOfLastImport = content.indexOf("\n", lastImportIdx + 1);
  if (lastImportIdx === -1) {
    console.log(`SKIP (no imports found): ${filename}`);
    continue;
  }
  content =
    content.slice(0, endOfLastImport + 1) +
    `import AppLayout from "@/components/AppLayout";\n` +
    content.slice(endOfLastImport + 1);

  // 2. Find the LAST "return (" in the file (the main component return)
  // We need to wrap its JSX in <AppLayout>
  const lines = content.split("\n");
  
  // Find the last top-level "return (" (2-space indent = component body)
  let mainReturnLine = -1;
  for (let i = lines.length - 1; i >= 0; i--) {
    if (/^  return \(/.test(lines[i])) {
      mainReturnLine = i;
      break;
    }
  }

  if (mainReturnLine === -1) {
    console.log(`SKIP (no return found): ${filename}`);
    continue;
  }

  // Find the closing ); of that return block
  // Walk forward from mainReturnLine, tracking brace/paren depth
  let depth = 0;
  let closingLine = -1;
  for (let i = mainReturnLine; i < lines.length; i++) {
    const line = lines[i];
    for (const ch of line) {
      if (ch === "(" || ch === "<") depth++;
      if (ch === ")" || ch === ">") depth--;
    }
    // The closing ); line has depth back to 0 or below after the return (
    if (i > mainReturnLine && /^\s*\);/.test(lines[i])) {
      closingLine = i;
      break;
    }
  }

  if (closingLine === -1) {
    console.log(`SKIP (no closing ); found): ${filename}`);
    continue;
  }

  // Insert <AppLayout> after "return (" and </AppLayout> before ");"
  lines[mainReturnLine] = lines[mainReturnLine].replace("return (", "return (\n    <AppLayout>");
  lines[closingLine] = lines[closingLine].replace(");", "    </AppLayout>\n  );");

  const newContent = lines.join("\n");
  fs.writeFileSync(filePath, newContent, "utf8");
  console.log(`FIXED: ${filename}`);
}

console.log("\nDone.");
